spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -615,97 +639,118 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
+            metadata = dict(Ecephys=dict())
+            if not write_as_lfp:
+                metadata['Ecephys'].update(
+                    ElectricalSeries=dict(raw_defaults)
+                )
+            else:
+                metadata['Ecephys'].update(
+                    LFPElectricalSeries=dict(lfp_defaults)
                 )
+        assert ('ElectricalSeries' in metadata['Ecephys']
+                and isinstance(metadata['Ecephys']['ElectricalSeries'], dict)) \
+            or ('LFPElectricalSeries' in metadata['Ecephys']
+                and isinstance(metadata['Ecephys']['LFPElectricalSeries'], dict)), \
+            "Expected either metadata['Ecephys']['ElectricalSeries'] or metadata['Ecephys']['LFPElectricalSeries'] " \
+            "to be a dictionary!"
+
+        if not write_as_lfp:
+            es_name = metadata['Ecephys']['ElectricalSeries'].get('name', raw_defaults['name'])
+            assert es_name not in nwbfile.acquisition, \
+                f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
+        else:
+            es_name = metadata['Ecephys']['LFPElectricalSeries'].get('name', lfp_defaults['name'])
+            ecephys_mod = check_module(
+                nwbfile,
+                'ecephys',
+                "Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
             )
-        assert isinstance(metadata['Ecephys']['ElectricalSeries'], dict), \
-            "Expected metadata['Ecephys']['ElectricalSeries'] to be a dictionary!"
-
-        es_name = metadata['Ecephys']['ElectricalSeries'].get('name', defaults['name'])
+            if 'LFP' not in ecephys_mod.data_interfaces:
+                ecephys_mod.add_data_interface(LFP(name='LFP'))
         channel_ids = recording.get_channel_ids()
-        if es_name not in nwbfile.acquisition:
-            table_ids = [list(nwbfile.electrodes.id[:]).index(id) for id in channel_ids]
-            electrode_table_region = nwbfile.create_electrode_table_region(
-                region=table_ids,
-                description="electrode_table_region"
+
+        table_ids = [list(nwbfile.electrodes.id[:]).index(id) for id in channel_ids]
+        electrode_table_region = nwbfile.create_electrode_table_region(
+            region=table_ids,
+            description="electrode_table_region"
+        )
+
+        # channels gains - for RecordingExtractor, these are values to cast traces to uV
+        # for nwb, the conversions (gains) cast the data to Volts
+        # To get traces in Volts = data*channel_conversion*conversion
+        gains = np.squeeze([recording.get_channel_gains(channel_ids=[ch])
+                            if 'gain' in recording.get_channel_property_names(channel_id=ch) else 1
+                            for ch in channel_ids])
+        if len(np.unique(gains)) == 1:  # if all gains are equal
+            scalar_conversion = np.unique(gains)[0] * 1e-6
```

Comment:
@CodyCBakerPhD is this to convert to uV? I would assume that gains already convert from int16 to uV...I would use gains as they are

Reply:
It's to convert from uV to V, V is the nwb ElectricalSeries standard unit. If extra conversion values are necessary (in that other line you commented on) for int16 translation, it is  assumed that from spikeextractors, those factors convert it from int16 to uV.

Reply:
Right! There's a minus ;)